nexus/src/app/vpc.rs

```diff
@@ -166,13 +173,49 @@ impl super::Nexus {
                 }
                 SubnetError::External(e) => e,
             })?;
-        let rules = db::model::VpcFirewallRule::vec_from_params(
+
+        // Customize the default firewall rules for this VPC.
+        let mut rules = db::model::VpcFirewallRule::vec_from_params(
             authz_vpc.id(),
             defaults::DEFAULT_FIREWALL_RULES.clone(),
         );
+        for rule in rules.iter_mut() {
+            for target in rule.targets.iter_mut() {
+                match target.0 {
+                    external::VpcFirewallRuleTarget::Vpc(ref mut name)
+                        if name.as_str() == "default" =>
+                    {
+                        *name = params.identity.name.clone()
+                    }
+                    _ => (),
```

Comment:
1dc02d3b now raises an error for these cases, which should never occur (if they do, it's because we made a mistake in specifying `DEFAULT_FIREWALL_RULES`).